### DISCUSSION TITLE
How to update the nodeType dynamically?

### DISCUSSION THREAD

**Discussion by samuellawerentz** [xyflow/xyflow#4449] [Q&A] [answered]
I want to update the nodeType when user clicks on some button in the UI. Can I do `updateNode` from `useReactFlow()`?

I tried this, but the node type is not getting updated. Is this the right way? Thanks!

```js
<button
  onClick={() => {
    updateNode('node-id', {
      type: 'new-node',
       data: {},
     })
   }}
>

**Comment by bcakmakoglu** [comment, marked as answer]
[Works fine for me in a sandbox.](https://codesandbox.io/p/sandbox/small-dust-tn3wyn?file=%2Fsrc%2FApp.tsx%3A21%2C21)

Are you sure that the `new-node` is even a node type you have defined?

**Comment by samuellawerentz** [comment]
Thank you for the quick reply! Yes, it worked fine! 

There was a issue with the state management. The node that I was updating and the node that was rendered were different. Thanks much!
